addons/mail/static/tests/helpers/mock_server/models/discuss_channel.js

```diff
@@ -595,6 +599,49 @@ patch(MockServer.prototype, {
         );
         return this._mockDiscussChannelChannelInfo([id])[0];
     },
+    /**
+     * Simulates `_channel_basic_info` on `discuss.channel`.
+     *
+     * @private
+     * @param {integer} id
+     * @returns {Object[]}
+     */
+    _mockDiscussChannelBasicInfo(id) {
+        const [channel] = this.getRecords("discuss.channel", [["id", "=", id]]);
+        const messages = this.getRecords("mail.message", [
+            ["model", "=", "discuss.channel"],
+            ["res_id", "=", channel.id],
+        ]);
+        const [group_public_id] = this.getRecords("res.groups", [
+            ["id", "=", channel.group_public_id],
+        ]);
+        const messageNeedactionCounter = this.getRecords("mail.notification", [
+            ["res_partner_id", "=", this.pyEnv.currentPartnerId],
+            ["is_read", "=", false],
+            ["mail_message_id", "in", messages.map((message) => message.id)],
+        ]).length;
+        const res = assignDefined({}, channel, [
+            "allow_public_upload",
+            "avatarCacheKey",
+            "create_uid",
+            "defaultDisplayMode",
+            "description",
+            "group_based_subscription",
+            "id",
+            "name",
+            "uuid",
+        ]);
+        Object.assign(res, {
+            channel_type: channel.channel_type,
+            memberCount: this.pyEnv["discuss.channel.member"].searchCount([
+                ["channel_id", "=", channel.id],
+            ]),
+            message_needaction_counter: messageNeedactionCounter,
```

Comment:
Make sure it has the same keys as the python counter-part, `message_needaction_counter` is user-specific and should not be moved to basic info